src/borg/archive.py

```diff
@@ -519,7 +519,7 @@ def extract_helper(self, dest, item, path, stripped_components, original_path, h
                 # assign chunks to this item, since the item which had the chunks was not extracted
                 item.chunks = chunks
         yield hardlink_set
-        if not hardlink_set and hardlink_masters:  # 2nd term, is it correct/needed?
```

Comment:
Ah right. So we just have to check here whether it is empty, because if it is indeed empty then there cannot be a master to update and the code path above cannot have extracted a slave as a master.

I created ticket #2388 for documenting these issues.